### ISSUE TITLE
'Permission denied' error on .ddev/ssh-auth-compose.yaml when running 'ddev start'

### ISSUE THREAD

**Issue by opdavies** [drud/ddev#2149]
**Describe the bug**
On a fresh Ubuntu installation with DDEV v1.13.1, I get the following error after running `ddev start`:

> Failed to start engineerlive: failed to start ddev-ssh-agent: Failed to run docker-compose [-f /home/opdavies/.ddev/ssh-auth-compose.yaml -p ddev-ssh-agent up --build --force-recreate -d], err='exit status 1', stdout='', stderr='.IOError: [Errno 13] Permission denied: '/home/opdavies/.ddev/ssh-auth-compose.yaml'' 

**Version and configuration information (please complete the following information):**
 - Host computer OS and Version: Ubuntu 18.04.4 LTS
 - Docker version information: 19.03.8
 - ddev version information v1.13.1
 - Standard generated config.yaml

**Additional context**

I also have the same issue when running the underlying command (`docker-compose -f /home/opdavies/.ddev/ssh-auth-compose.yaml -p ddev-ssh-agent up --build --force-recreate -d`).

I'm able to view the contents of that file though with `cat` or `head` with no issue.

**Comment by rfay** [comment]
You have probably run ddev as root at some time, and the ~/.ddev directory is not owned by you. 

I think if you `ls -lR ~/.ddev` you'll see that the files there are not owned by you, they're probably owned by root. 

`sudo chown -R opdavies:opdavies ~/.ddev` will probably fix it.

**Comment by opdavies** [comment]
Thanks for the quick reply, @rfay.

I've checked and the directory is owned by my user. I've `chown`-ed it just in case, but I still get the same error.

**Comment by rfay** [comment]
Please show the results of `id` and of `ls -lR ~/.ddev`

**Comment by rfay** [comment]
Hi @opdavies - haven't heard back from you. Hope you figured this out. I'll close it soon if I don't hear back. If you're still having trouble, please do show the results of `id` and of `ls -lR ~/.ddev`

**Comment by opdavies** [comment]
@rfay Here you go:

```
~ $ id
uid=501(opdavies) gid=1000(opdavies) groups=1000(opdavies),4(adm),24(cdrom),27(sudo),30(dip),46(plugdev),116(lpadmin),126(sambashare),999(docker)
                                                                                                                                                
~ $ ls -lR ~/.ddev 
/home/opdavies/.ddev:
total 8
-rw-r--r-- 1 opdavies opdavies 1222 Apr 10 09:22 global_config.yaml
-rw-r--r-- 1 opdavies opdavies  710 Apr 10 09:22 ssh-auth-compose.yaml
```

**Comment by rfay** [comment]
Well... it sure looks like your ssh-auth-compose got created just fine on April 10... So I guess you got past this?

**Comment by opdavies** [comment]
No, I'm still getting the error message,

**Comment by rfay** [comment]
What's the timestamp on your ~/.ddev/ssh-auth-compose.yaml ? `ls -l ~/.ddev/ssh-auth-compose.yaml` ? Does it always correspond to the last time you did something with ddev? When does it change? *something* is able to access it!

I should have asked you for `ls -ld ~/.ddev` as well, thanks. You may want to also check `ls -ld ~`

I guess if I were you, and I checked to make sure that my home directory permissions were correct (any changes there), I'd just `ddev poweroff` and `rm -r ~/.ddev`.  Let it all get recreated. Essentially something is wrong with your homedir permissions or ~/.ddev perms. There's nothing ddev can do about that. 

I would assume that if you `cat >>~/.ddev/ssh-auth-compose.yaml` you will also get the same permission error.

**Comment by rfay** [comment]
No response again, so closing. Happy to continue the conversation as always when you get back to this.

**Comment by froemken** [comment]
I just have the same problem with DDEV v1.16.7

I have installed a fresh Ubuntu Server System 20.10. While install process of Ubuntu I get a list of installable Server tools. I have chosen docker, nextcloud and microk8s. But this combination results in various problems:

docker was installed as snap in /usr/bin/snap. Interesting docker-compose is located in /usr/bin/ and is linked to /usr/bin/docker.compose and that is again linked to /usr/bin/snap.

It is not possible to remove docker-compose: sudo snap remove docker-compose
As it is part of the docker snap.

I have removed docker completely and installed docker and docker-compose my known way with docker repos. Now, the network_default of DDEV could be created successfully, but port 80 is blocked.

I have removed nextcloud and microk8s, too, and now DDEV can start my project.

Maybe this helps others to solve that problem.

Stefan
